frontend/src/views/WheresMyGene/components/HeatMap/components/Chart/index.tsx

```diff
@@ -361,56 +344,70 @@ export default memo(function Chart({
 
   const tooltipClasses = useMemo(() => ({ tooltip: tooltipCss }), []);
 
+  const tooltipPopperProps = useMemo(() => {
+    return {
+      anchorEl: {
+        getBoundingClientRect: () => ({
+          bottom: cursorOffset[1],
+          height: 0,
+          left: cursorOffset[0],
+          right: cursorOffset[0],
+          toJSON: noop,
+          top: cursorOffset[1],
+          width: 0,
+          x: cursorOffset[0],
+          y: cursorOffset[1],
+        }),
+      },
+      modifiers: [
+        {
+          name: "offset",
+          options: {
+            offset: [0, 5],
+          },
+        },
+      ],
+    };
+  }, [cursorOffset]);
+
+  const handleMouseMove = useCallback(() => {
+    clearInterval(handleDotHoverAnalytic);
+    if (tooltipContent?.props.data) {
+      handleDotHoverAnalytic = setTimeout(() => {
+        track(EVENTS.WMG_HEATMAP_DOT_HOVER);
+      }, 2 * 1000);
+    }
+  }, [tooltipContent]);
+
+  const { chartData, cellTypeMetadata, geneNames } = chartProps || {};
+
+  const itemStyle = useChartItemStyle(isScaled);
+
   return (
     <Tooltip
       width="wide"
       classes={tooltipClasses}
       title={tooltipContent || <>No data</>}
       leaveDelay={0}
       placement="right-end"
-      onMouseMove={() => {
-        clearInterval(handleDotHoverAnalytic);
-        if (tooltipContent?.props.data) {
-          handleDotHoverAnalytic = setTimeout(() => {
-            track(EVENTS.WMG_HEATMAP_DOT_HOVER);
-          }, 2 * 1000);
-        }
-      }}
-      PopperProps={{
-        anchorEl: {
-          getBoundingClientRect: () => ({
-            bottom: cursorOffset[1],
-            height: 0,
-            left: cursorOffset[0],
-            right: cursorOffset[0],
-            toJSON: noop,
-            top: cursorOffset[1],
-            width: 0,
-            x: cursorOffset[0],
-            y: cursorOffset[1],
-          }),
-        },
-        modifiers: [
-          {
-            name: "offset",
-            options: {
-              offset: [0, 5],
-            },
-          },
-        ],
-      }}
+      onMouseMove={handleMouseMove}
+      PopperProps={tooltipPopperProps}
     >
-      <ChartContainer
+      <RawChart
         height={heatmapHeight}
         width={heatmapWidth}
         ref={ref}
         id={`${tissue.replace(/\s+/g, "-")}-chart`}
-        onMouseLeave={() => {
-          // Handles race condition when a timeout is set after clearing
-          setTimeout(() => {
-            clearTimeout(handleDotHoverAnalytic);
-          }, 100);
-        }}
+        onMouseLeave={handleMouseLeave}
+        data={chartData}
+        yAxisData={cellTypeMetadata}
+        xAxisData={geneNames}
+        echartsRendererMode={echartsRendererMode}
+        onChartMouseMove={handleChartMouseMove}
+        encode={CHART_DATA_TO_AXIS_ENCODING}
+        itemStyle={itemStyle}
+        symbolSize={symbolSize}
+        grid={grid}
```

Comment:
since `RawChart` now handles heatmap generation, we use the new simplified interface with echarts without needing to know how echarts works

Reply:
Great question 😭 

So the new component is also named `Chart`, so in order to avoid name collision, I named it RawChart when importing it into the parent `Chart` component here:

```ts
import RawChart from "./components/Chart";
```

The naming dilemma I have is added here:

https://github.com/chanzuckerberg/single-cell-data-portal/pull/4844/files#r1195710093